src-tauri/src/document/types.rs

```diff
@@ -288,34 +295,39 @@ pub struct Sample {
     pub fx: [f64; 4],
     pub fy: [f64; 4],
 }
+
 #[derive(Serialize, Deserialize, Clone)]
-pub struct ChoreoPath<T> {
+pub struct ChoreoPath<T: SnapshottableType> {
     pub waypoints: Vec<Waypoint<T>>,
     pub constraints: Vec<Constraint<T>>,
 }
 
-impl ChoreoPath<Expr> {
+impl<T: SnapshottableType> ChoreoPath<T> {
     pub fn snapshot(&self) -> ChoreoPath<f64> {
         ChoreoPath {
             waypoints: self.waypoints.iter().map(Waypoint::snapshot).collect(),
             constraints: self.constraints.iter().map(Constraint::snapshot).collect(),
         }
     }
 }
+
 #[derive(Serialize, Deserialize, Clone)]
-#[allow(non_snake_case)]
+#[serde(rename_all = "camelCase")]
 pub struct Output {
     pub waypoints: Vec<f64>,
     pub samples: Vec<Vec<Sample>>,
 
-    pub useModuleForces: bool,
+    pub use_module_forces: bool,
 }
+
 #[derive(Serialize, Deserialize, Clone)]
+#[allow(clippy::struct_field_names)]
 pub struct Traj {
     pub name: String,
     pub version: String,
     pub path: ChoreoPath<Expr>,
     // Captures the path that formed the current generated trajectory
     pub snapshot: Option<ChoreoPath<f64>>,
+    // TODO: maybe rename to `output`, this is a breaking change for frontend though
```

Comment:
Now is a good time to be breaking things...

Reply:
i wanted this to be a purely non functional pr so it can get merged quickly